Fantasy-Quest/Assets/Scripts/Cutscene/CutsceneInteractable.cs

```diff
@@ -0,0 +1,37 @@
+using Interaction.Item;
+using Sirenix.OdinInspector;
+using UnityEngine;
+
+namespace Cutscene
+{
+    //Cutscene played depends on canCatInteract condition(Can = E, Can't = 1)
+    [AddComponentMenu("Scripts/Cutscene/Cutscene.CutsceneInteract")]
+    internal class CutsceneInteractable : MonoBehaviour, IInteractable
+    {
+        [Required]
+        [SerializeField]
+        private Start cutsceneStarter;
+
+        [SerializeField]
+        private bool canCatInteract;
+        bool IInteractable.CanCatInteract => canCatInteract;
+
+        public void InteractionByCat()
+        {
+            if (canCatInteract)
+            {
+                cutsceneStarter.StartCutscene();
+                Destroy(this);
```

Comment:
Better use SetActive(false) to turn off GameObject or use enabled = false to turn Component off